src/game/game.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@nestjs/common';
+import { GameRepository } from './game.repository';
+import { PlayerService } from 'src/player/player.service';
+
+@Injectable()
+export class GameService {
+  constructor(
+    private gameRepository: GameRepository,
+    private playerService: PlayerService
+  ) {}
+
+  async getAllVisibleGames() {
+    const games = await this.gameRepository.findMany({
+      where: { status: 'LOBBY' },
+      include: {
+        players: {
+          include: { user: { select: { nickname: true } } },
+        },
+      },
+    });
+    return games;
+  }
+
+  async getCurrentGame(gameId: string) {
+    return this.gameRepository.findFirst({
+      where: { id: gameId },
+      include: {
+        players: { include: { user: { select: { nickname: true } } } },
+      },
+    });
+  }
+
+  async joinGame(gameId: string, userId: string) {
+    const game = await this.gameRepository.findFirst({
+      where: { id: gameId },
+      include: { players: true },
+    });
+
```

Comment:
It could be that game is not exist  with this 'id' , findFirst method will return null in this case. So, you should check for this value and throw an HttpException